ci-operator/step-registry/cucushift/upgrade/toimage/cucushift-upgrade-toimage-commands.sh

```diff
@@ -383,7 +370,18 @@ function check_upgrade_status() {
             echo -e "Upgrade stuck at updating RHEL worker, need to run the RHEL worker upgrade later...\n\n"
             return 0
         fi
-        dump_status_if_unexpected
+        # assuming oc to equal TARGET_MINOR_VERSION, log abnormal "upgrade status" if available (version 4.16+)
+        if [[ -n "${TARGET_MINOR_VERSION}" ]] && [[ "${TARGET_MINOR_VERSION}" -ge "16" ]] ; then
+            # capture "update health" section of status, ignoring the title and the last line
+            # whenever an error is present, "Run with --details=health" is the last line we should remove.
+            # with no error, it'll remove "Update is proceeding well" resulting in empty $stat that we simply ignore
+            stat="$(env OC_ENABLE_CMD_UPGRADE_STATUS=true oc adm upgrade status 2>&1 | sed -ne "/= Update Health =/,$ p" | sed '1d;2d;$d' || true)"
```

Comment:
I have some concerns of only printing unhealthy message, sometimes the whole output from `oc adm upgrade status` can help us find out some useful info, for example, https://issues.redhat.com/browse/OCPQE-22351 was found when checking the complete output of `oc adm upgrade status`

Reply:
that one is a dilemma as well. firstly, after upgrade failure, `function debug` will print the full version, +including details.   
during upgrade, we can print either the full long message, or the Update Health. both options have pros and cons, so I have asked [in the thread](https://redhat-internal.slack.com/archives/CF8SMALS1/p1719907992107919?thread_ts=1719889316.376549&cid=CF8SMALS1) for our teammates opinion, and both @JianLi-RH as well as @jiajliu commented they prefer shorter log with only "Update Health" section

Reply:
Actually I am thinking if we can  increase the intervals to save some output, but keep a competed output from the command when it was executed, also adding some empty lines as separators to make the log easier to read.

Reply:
good point about monitoring functionality. this was the original intention (alongside obviously providing insights) to implement upgrade status into the CI. however as for increasing the interval, this would have an obvious negative effect of missing certain temporary conditions, because upgrade status does not save information but only prints live

Reply:
Imo, since we're anyway skipping the happy state status of "Update is proceeding well" as well as the duplicates, so a log being too long should not be a concern after this PR, whatever we decide to print the full message or the health only, in any case should be a good idea to actually decrease intervals!

when I tested the status command while working at the epic, I even used a zero interval, but a bit more aggressive filter to ignore all time and percentage differences, and ended up with acceptable sized log.

Reply:
For me, the original full output from `oc adm upgrade status` is not a pain point for me, after all, I used the output (even it is a healthy message) to find out some automation issues. And standing from checking the functionality of the command, I prefer to keep a full output. From my side, I think the only enhancement is to make the whole log output easier to read, but not limit the log's length.

Reply:
> From my side, I think the only enhancement is to make the whole log output easier to read, but not limit the log's length. cc @JianLi-RH to think about it.

Agree with @jianlinliu , we can add separators between each upgrade status output.